okhttp/src/commonMain/kotlin/okhttp3/internal/-ResponseCommon.kt

```diff
@@ -30,9 +31,44 @@ import okhttp3.internal.http.HTTP_MULT_CHOICE
 import okhttp3.internal.http.HTTP_PERM_REDIRECT
 import okhttp3.internal.http.HTTP_SEE_OTHER
 import okhttp3.internal.http.HTTP_TEMP_REDIRECT
-
 import okio.Buffer
 import okio.IOException
+import okio.Source
+import okio.Timeout
+import okio.buffer
+
+class UnreadableResponseBody(
+  private val mediaType: MediaType?,
+  private val contentLength: Long,
+): ResponseBody(), Source {
+  override fun contentType() = mediaType
+  override fun contentLength() = contentLength
+  override fun source() = buffer()
+
+  override fun read(sink: Buffer, byteCount: Long): Long {
+    throw IllegalStateException("""
```

Comment:
I agree it's a logic bug, so IllegalStateException is warranted.  My first reaction, was IOException is safer, but as I'm typing this I realise that would be confusing and probably handled in app code with a retry.